### PR TITLE
S2178 - nested transaction fix

### PR DESCRIPTION
**Pull request by pdowler**

### CODE REVIEW

File: caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/action/PutAction.java [hunk 1]
```diff
@@ -109,40 +107,42 @@ public void doAction() throws Exception {
         }
 
         validate(obs);
-        ReadAccessTuplesGenerator ratGenerator = getReadAccessTuplesGenerator(getCollection(), getReadAccessDAO(), getReadAccessGroupConfig());
+        final ReadAccessTuplesGenerator ratGenerator = getReadAccessTuplesGenerator(getCollection(), getReadAccessDAO(), getReadAccessGroupConfig());
```
Comment (yeunga) [inline caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/action/PutAction.java]: Add "final" to ratGenerator in PostAction as well?

File: caom2persistence/src/main/java/ca/nrc/cadc/caom2/persistence/DatabaseTransactionManager.java [hunk 2]
```diff
@@ -90,10 +89,20 @@
     private static final Logger log = Logger.getLogger(DatabaseTransactionManager.class);
 
     private DataSourceTransactionManager writeTxnManager;
-    private final TransactionDefinition def = new DefaultTransactionDefinition();
+    private final TransactionDefinition defaultTxnDef = new DefaultTransactionDefinition();
     private final TransactionDefinition nested = new DefaultTransactionDefinition(TransactionDefinition.PROPAGATION_NESTED);
-    private final Deque<TransactionStatus> txn = new LinkedList<TransactionStatus>();
+    private final Deque<Txn> transactions = new LinkedList<>();
```
Comment (yeunga) [inline caom2persistence/src/main/java/ca/nrc/cadc/caom2/persistence/DatabaseTransactionManager.java]: Use "new LinkedList<Txn>()" to specify the type of elements in the LinkedList?

File: caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/action/PutAction.java (hunk 1 above)
Comment (pdowler) [reply caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/action/PutAction.java]: fixed

File: caom2persistence/src/main/java/ca/nrc/cadc/caom2/persistence/DatabaseTransactionManager.java (hunk 2 above)
Comment (pdowler) [reply caom2persistence/src/main/java/ca/nrc/cadc/caom2/persistence/DatabaseTransactionManager.java]: fixed

**Pull request merged**